### PR TITLE
chore: remove redundant config for Lua

### PR DESCRIPTION
**Pull request by hezhizhen**
There are no different configs for Lua, so I removed its section.

I also fix some typos.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -8,10 +8,6 @@ end_of_line = lf
 insert_final_newline = true
 charset = utf-8
 
-[*.lua]
```
Comment (glepnir) [inline .editorconfig:11]: why remove this from editorconfig.  this is useful for who use neovim builtin editorconfig.

Comment (hezhizhen) [reply .editorconfig:11]: I think there are no different configs for Lua, so I removed its section. Does the settings in [*] section not work for neovim builtin editorconfig?

Comment (polyzen) [reply .editorconfig:11]: Because it's covered already in the `[*]` block

Comment (glepnir) [reply .editorconfig:11]: right . it already in `*` block.

**Comment by glepnir** [comment]
LGTM

**Pull request merged**